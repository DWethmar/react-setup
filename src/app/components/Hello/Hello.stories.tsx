import React from 'react';
import { Hello } from './Hello';

export default {
    title: 'Hello',
    component: Hello,
};

export const Default = (): React.ReactElement => <Hello compiler="typescript" framework="react" />;
