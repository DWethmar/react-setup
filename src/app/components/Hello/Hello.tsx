import * as React from 'react';
import * as PropTypes from 'prop-types';

import './hello.scss';

export interface HelloProps {
    compiler: string;
    framework: string;
}

const Hello: React.StatelessComponent<HelloProps> = ({ compiler, framework }) => {
    return (
        <h1>
            Hello from {compiler} and {framework}!
        </h1>
    );
};

Hello.propTypes = {
    compiler: PropTypes.string.isRequired,
    framework: PropTypes.string.isRequired,
};

export { Hello };
