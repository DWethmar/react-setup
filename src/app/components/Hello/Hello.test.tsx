import React from 'react';
import { shallow } from 'enzyme';
import { Hello } from './Hello';

describe('Hello', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Hello compiler="typescript" framework="react" />);

        expect(component).toMatchSnapshot();
    });
});
