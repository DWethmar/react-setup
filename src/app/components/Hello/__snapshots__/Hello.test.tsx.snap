// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello should match snapshot 1`] = `
<h1>
  Hello from 
  typescript
   and 
  react
  !
</h1>
`;
